dnsapi/dns_wedos.sh

```diff
@@ -0,0 +1,366 @@
+#!/usr/bin/env sh
+
+# This script has been created at June 2020, based on knowledge base of wedos.com provider.
+# It is intended to allow DNS-01 challenges for acme.sh using wedos's WAPI using XML.
+
+# Author Michal Tuma <mxtuma@gmail.com>
+# For issues send me an email
+
+WEDOS_WAPI_ENDPOINT="https://api.wedos.com/wapi/xml"
+TESTING_STAGE=
+
+########  Public functions #####################
+
+#Usage: add  _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_wedos_add() {
+  fulldomain=$1
+  txtvalue=$2
+
+  WEDOS_Username="${WEDOS_Username:-$(_readaccountconf_mutable WEDOS_Username)}"
+  WEDOS_Wapipass="${WEDOS_Wapipass:-$(_readaccountconf_mutable WEDOS_Wapipass)}"
+  WEDOS_Authtoken="${WEDOS_Authtoken:-$(_readaccountconf_mutable WEDOS_Authtoken)}"
+
+  if [ "${WEDOS_Authtoken}" ]; then
+    _debug "WEDOS Authtoken was already saved, using saved one"
+    _saveaccountconf_mutable WEDOS_Authtoken "${WEDOS_Authtoken}"
+  else
+    if [ -z "${WEDOS_Username}" ] || [ -z "${WEDOS_Wapipass}" ]; then
+      WEDOS_Username=""
+      WEDOS_Wapipass=""
+      _err "You didn't specify a WEDOS's username and wapi key yet."
+      _err "Please type: export WEDOS_Username=<your user name to login to wedos web account>"
+      _err "And: export WEDOS_Wapipass=<your WAPI passwords you setup using wedos web pages>"
+      _err "After you export those variables, run the script again, the values will be saved for future"
+      return 1
+    fi
+
+    #build WEDOS_Authtoken
+    _debug "WEDOS Authtoken were not saved yet, building"
+    WEDOS_Authtoken=$(printf '%s' "${WEDOS_Wapipass}" | sha1sum | head -c 40)
+    _debug "WEDOS_Authtoken step 1, WAPI PASS sha1 sum: '${WEDOS_Authtoken}'"
+    WEDOS_Authtoken="${WEDOS_Username}${WEDOS_Authtoken}"
+    _debug "WEDOS_Authtoken step 2, username concat with token without hours: '${WEDOS_Authtoken}'"
+
+    #save details
+
+    _saveaccountconf_mutable WEDOS_Username "${WEDOS_Username}"
+    _saveaccountconf_mutable WEDOS_Wapipass "${WEDOS_Wapipass}"
+    _saveaccountconf_mutable WEDOS_Authtoken "${WEDOS_Authtoken}"
+  fi
+
+  if ! _get_root "${fulldomain}"; then
+    _err "WEDOS Account do not contain primary domain to fullfill add of ${fulldomain}!"
+    return 1
+  fi
+
+  _debug _sub_domain "${_sub_domain}"
+  _debug _domain "${_domain}"
+
+  if _wapi_row_add "${_domain}" "${_sub_domain}" "${txtvalue}" "300"; then
```

Comment:
what is the minimum value of the `ttl` value?  `300`?

Reply:
Yes, for WEDOS 300 is minimum.

Reply:
Do you feel something is wrong? Just using TTL 300 as default.